.github/workflows/_binary-build-linux.yml

```diff
@@ -11,6 +11,10 @@ on:
         required: true
         type: string
         description: The build environment
+      runs_on:
+          required: true
+          type: string
+          description: Hardware to run this job on. Valid values are linux.4xlarge, linux.4xlarge.nvidia.gpu, and linux.rocm.gpu
```

Comment:
I think @malfet idea was to refactor the ``./.github/workflows/_binary-build-linux.yml`` and bring required changes from ``./.github/workflows/_binary-build-aarch64.yml`` (they are almost the same) .

I see following changes would be needed
- Add ``ALPINE_IMAGE`` as a new param with default value "308535385114.dkr.ecr.us-east-1.amazonaws.com/tool/alpine" for all Linux job, for aarch64 job we would need to pass "arm64v8/alpine"
- Make changes to following steps: Chown workspace, Build PyTorch binary

Reply:
List of differences:
* Different Instances for runs_on (working a resolution using the template)
* Different Docker Images (working a resolution using the template)
* Aarch64 cannot use .github/actions/chown-workspace because the image is hardcoded there for x86_64 (no resolution yet, need time to see if I can pass the alpine ENV from the build workflow to the chown-workspace workflow, if you know this answer now, would save me time in testing it.)
* Different build scripts (x86_64 is using the main bash scripts where aarch64 uses aarch64 specific bash/python scripts, this will take a bit to add logic into run calls for the scripts with a possibility of affecting x86_64, which I already did with [faf9383](https://github.com/pytorch/pytorch/pull/104109/commits/faf9383d4a8929c0870d5d903ce7353e539c3818), and you called out in the review. Thank you for catching that)

Reply:
Looks like all the difference have been integrated into the main linux workflows.